src/shared/views/BuildingFootprint/FootprintDisplay.js

```diff
@@ -1,43 +1,101 @@
 // @flow
 import * as React from 'react';
-import { Image, StyleSheet, View } from 'react-native';
+import { Image, PanResponder, StyleSheet, View } from 'react-native';
+import { type PressEvent } from 'react-native/Libraries/Types/CoreEventTypes';
+import type {
+    GestureState,
+    PanResponderInstance,
+} from 'react-native/Libraries/Interaction/PanResponder';
 import { Path, Shape, Surface } from '@react-native-community/art';
 import tilebelt from '@mapbox/tilebelt';
 import type {
     BBOX,
     Point,
     Polygon,
-    ProjectType,
+    SingleImageryProjectType,
     BuildingFootprintTaskType,
     Tile,
 } from '../../flow-types';
 
 const GLOBAL = require('../../Globals');
 
-const tileSize = GLOBAL.SCREEN_WIDTH;
-
 const styles = StyleSheet.create({
     tileImg: {
-        height: tileSize,
         position: 'absolute',
-        width: tileSize,
     },
 });
 
 type Props = {
-    project: ProjectType,
+    nextTask: () => void,
+    previousTask: () => void,
+    project: SingleImageryProjectType,
     task: BuildingFootprintTaskType,
 };
 
 export default class FootprintDisplay extends React.Component<Props> {
+    panResponder: PanResponderInstance;
+
+    swipeThreshold: number;
+
+    // the imagery is always shown as a square, whose size is computed to fill
+    // in the screen as much as possible, while remaining narrower than the screen.
+    // we rely on flexbox to size the different components, which means that the
+    // first rendering is when we find out how much space we have for the imagery.
+    // so we first set this value to null, and update it to min(height, screenwidth)
+    // once flexbox has given us a height. Only then can we pull imagery
+    imagerySize: number;
+
+    constructor(props: Props) {
+        super(props);
+        // swipeThreshold defines how much movement is needed to start considering the event
+        // as a swipe. This used to be a fixed value, we now link it to screen size (through the tile size)
+        // so that it should work across screen densities.
+        this.swipeThreshold = GLOBAL.TILE_SIZE * 0.02;
+        this.panResponder = PanResponder.create({
+            onMoveShouldSetPanResponder: this.handleMoveShouldSetPanResponder,
+            onMoveShouldSetPanResponderCapture: this
+                .handleMoveShouldSetPanResponder,
+            onPanResponderRelease: this.handlePanResponderEnd,
+        });
+        this.imagerySize = 0;
+    }
+
+    // $FlowFixMe
```

Comment:
what is this comment about?

Reply:
This tells `flow` to ignore the next line because I couldn't find the correct type definition for the event. I'll look a bit harder to fix it.